[aotinductor] Handle scalar as the input argument

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #111973
* #112021
* #111823

Summary: For https://github.com/pytorch/pytorch/issues/111830

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

desertfire committed:
[aotinductor] Handle scalar as the input argument

Summary: For https://github.com/pytorch/pytorch/issues/111830

[ghstack-poisoned]

chenyang78 commented on test/inductor/test_aot_inductor.py:
```diff
@@ -972,6 +993,7 @@ class AOTInductorTestABICompatibleCuda(TestCase):
     "abi_compatible_cuda",
     # test_failures, xfail by default, set is_skip=True to skip
     {
+        "test_arange_with_start": TestFailure(("abi_compatible_cuda",)),
```
Wondering why this test would fail in the abi_compatible_cuda mode?

desertfire replied:
because it requires `aoti_torch_normal_functional`

desertfire committed:
Update on "[aotinductor] Handle scalar as the input argument"


Summary: For https://github.com/pytorch/pytorch/issues/111830

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

desertfire commented on test/inductor/test_aot_inductor.py:
```diff
@@ -93,6 +93,11 @@ def run_compiled(cls, optimized, exported, example_inputs):
         flat_example_inputs = fx_pytree.tree_flatten_spec(
             (example_inputs, {}), exported.call_spec.in_spec
         )
+        # For scalar inputs, we need to wrap them into tensors
```
@angelayi , I wonder if there is any existing util function in export that does this?

angelayi replied:
Wait sorry, I'm confused why we need to wrap them as tensors. Does AOTInductor assume all inputs are tensors? I feel like that should be fixed then, because isn't there work to support symints as inputs?

desertfire replied:
The AOTInductor generated code has a uniform entry function signature, which expects a vector of tensors. It's more flexible in the JIT mode where Python tuple can contain different types.  cc @ezyang

ezyang replied:
Ultimately, AOTInductor should support int inputs, if only for efficiency reasons, because you are going to expect a nontrivial number of size-like arguments, for which it is pointless to wrap them into a full blown Tensor, just to immediately unwrap them again.

ezyang replied:
cc @swolchok

desertfire committed:
Update on "[aotinductor] Handle scalar as the input argument"


Summary: For https://github.com/pytorch/pytorch/issues/111830

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

desertfire committed:
Update on "[aotinductor] Handle scalar as the input argument"


Summary: For https://github.com/pytorch/pytorch/issues/111830

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]
